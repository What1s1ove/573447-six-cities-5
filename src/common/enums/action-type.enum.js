const PlacesDataActionType = {
  LOAD_OFFERS: `LOAD_OFFERS`,
};

export {PlacesDataActionType};
