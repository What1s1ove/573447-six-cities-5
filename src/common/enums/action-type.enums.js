const PlacesDataActionType = {
  LOAD_OFFERS: `LOAD_OFFERS`,
  UPDATE_OFFER: `UPDATE_OFFER`,
};

const PlaceActionType = {
  LOAD_OFFER: `LOAD_OFFER`,
  LOAD_REVIEWS: `LOAD_REVIEWS`,
  LOAD_SIMILAR_OFFERS: `LOAD_SIMILAR_OFFERS`,
  UPDATE_SIMILAR_OFFER: `UPDATE_SIMILAR_OFFER`,
};

const FavoritesActionType = {
  LOAD_FAVORITES: `LOAD_FAVORITES`,
  UPDATE_FAVORITE: `UPDATE_FAVORITE`,
};

const UserActionType = {
  SET_AUTH_STATUS: `SET_AUTH_STATUS`,
  SET_USER: `SET_USER`,
  REDIRECT_TO_ROUTE: `REDIRECT_TO_ROUTE`,
};

export {
  PlacesDataActionType,
  PlaceActionType,
  UserActionType,
  FavoritesActionType,
};
