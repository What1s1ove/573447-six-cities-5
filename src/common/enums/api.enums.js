const HttpCode = {
  SUCCESS: 200,
  UNAUTHORIZED: 401,
};

const ApiRoute = {
  LOGIN: `/login`,
  HOTELS: `/hotels`,
  FAVORITE: `/favorite`,
  COMMENTS: `/comments`,
};

export {HttpCode, ApiRoute};
