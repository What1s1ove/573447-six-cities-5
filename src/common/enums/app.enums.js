const AppRoute = {
  FAVORITES: `/favorites`,
  LOGIN: `/login`,
  MAIN: `/`,
  OFFER: `/offer`,
  OFFER_ID: `/offer/:id`,
};

const AppConfig = {
  OFFERS_COUNT: 40,
  REVIEWS_COUNT: 5,
};

const SortType = {
  POPULAR: `Popular`,
  TO_HIGHT: `Price: low to high`,
  TO_LOW: `Price: high to low`,
  TOP_RATE: `Top rated first`,
};

export {AppRoute, AppConfig, SortType};
