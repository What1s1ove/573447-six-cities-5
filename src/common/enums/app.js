const AppRoute = {
  MAIN: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer`,
  OFFER_ID: `/offer/:id`,
};

export {AppRoute};
