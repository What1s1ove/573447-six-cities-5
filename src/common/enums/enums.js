export * from './app.enums';
export * from './offers.enums';
export * from './api.enums';
export * from './action-type.enum';
