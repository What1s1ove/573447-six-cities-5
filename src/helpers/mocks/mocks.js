export * from './generate-offer.helper';
export * from './generate-offers.helper';
export * from './generate-random-id.helper';
