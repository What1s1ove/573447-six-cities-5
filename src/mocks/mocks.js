export * from './offers/offers';
export * from './user/user';
export * from './reviews/reviews';
export * from './city/city';
export * from './store/store';
