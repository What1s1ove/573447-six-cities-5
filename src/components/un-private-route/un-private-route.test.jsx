import * as React from 'react';
import {render} from 'enzyme';
import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';
import {store} from '~/mocks/mocks';
import {AppRoute} from '~/common/enums/enums';
import UnPrivateRoute from './un-private-route';

describe(`UnPrivateRoute component`, () => {
  it(`should render UnPrivateRoute component`, () => {
    const component = render(
        <Provider store={store}>
          <MemoryRouter>
            <UnPrivateRoute path={AppRoute.LOGIN} component={null} />
          </MemoryRouter>
        </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
