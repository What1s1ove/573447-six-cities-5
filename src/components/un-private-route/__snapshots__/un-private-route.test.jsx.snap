// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnPrivateRoute component should render UnPrivateRoute component 1`] = `null`;
